src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
```

Comment:
The stack array should be of type T instead of Object for type safety. Replace `private Object[] stack;` with `private T[] stack;` and adjust the rest of your code accordingly.